src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -926,35 +1112,57 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
    * returns true if the thread that ended was a normally running task thread.
    * This should be used to decide if wake_up_threads_after_task_finishes is called or not.
    */
-  bool remove_thread_association(long thread_id, const std::unique_lock<std::mutex>& lock)
+  bool remove_thread_association(long thread_id, long remove_task_id, const std::unique_lock<std::mutex>& lock)
```

Comment:
Nit: Please put `const` after type (west const) to be consistent with the current repository style.